### ISSUE TITLE
Database Sharding Policy in Reverse Engineer

### ISSUE THREAD

**Issue by mawarren642** [microsoft/delta-kusto#95]
When downloading a schema as a .kql file we obtain its database policy sharding commands, i.e.

```
.alter database ['demo'] policy sharding
{
  "MaxRowCount": 750000,
  "MaxExtentSizeInMb": 1024,
  "MaxOriginalSizeInMb": 2048,
  "UseShardEngine": true,
  "ShardEngineMaxRowCount": 1048576,
  "ShardEngineMaxExtentSizeInMb": 8192,
  "ShardEngineMaxOriginalSizeInMb": 3072
}
```
It seems that the line `"UseShardEngine": true,` is somewhat useless-- according to the [public documentation we're consulting](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/management/shardingpolicy) this value cannot be changed after database creation, and trying to reverse engineer it into any existing database where the 'UseShardEngine' value is different than what is in the schema causes the following error:

```{
    "error": {
        "code": "BadRequest",
        "message": "Request is invalid and cannot be executed.",
        "@type": "Kusto.DataNode.Exceptions.InvalidPolicyOperationException",
        "@message": "Error during execution of a policy operation: UseShardEngine can not be set, if shard engine is not enabled",
        "@context": {
            "timestamp": "2022-05-18T18:37:18.5938655Z",
            "serviceAlias": "BALROGDEVKUSTO",
            "machineName": "KEngine000000",
            "processName": "Kusto.WinSvc.Svc",
            "processId": 3548,
            "threadId": 704,
            "appDomainName": "Kusto.WinSvc.Svc.exe",
            "clientRequestId": "KE.RunCommand;dcd356bb-eb5a-45e7-8ee1-23f0280c787e",
            "activityId": "49d01113-e782-4bd9-b344-4a3933b3ad48",
            "subActivityId": "742ab8d1-ba5f-4db2-a652-0d5da42ea1bf",
            "activityType": "DN.AdminCommand.DatabaseScopedDataShardingPolicyAlterCommand",
            "parentActivityId": "32aef848-6f6a-487f-829e-ae71a2bfb7d1",
            "activityStack": "(Activity stack: CRID=KE.RunCommand;dcd356bb-eb5a-45e7-8ee1-23f0280c787e ARID=49d01113-e782-4bd9-b344-4a3933b3ad48 > DN.Admin.Client.ExecuteControlCommand/8375c0e8-e25c-4788-9821-893c959482a9 > P.WCF.Service.ExecuteControlCommand..IInterNodeCommunicationAdminContract/232e1f1a-896c-429e-b56e-d9383db8f1da > DN.FE.ExecuteControlCommand/32aef848-6f6a-487f-829e-ae71a2bfb7d1 > DN.AdminCommand.DatabaseScopedDataShardingPolicyAlterCommand/742ab8d1-ba5f-4db2-a652-0d5da42ea1bf)"
        },
        "@permanent": true
    }
}
```

We have automated our pipeline to comment out the command in the schema, but I think it would be a good idea to simply make the delta kusto reverse engineer command suppress this error into a warning and continue, as there does not seem to be a scenario where issuing the command at all would accomplish anything.

**Comment by vplauzon** [comment]
Good point.  We'll fix it.

**Comment by vplauzon** [comment]
Fixed in [0.10.4](https://github.com/microsoft/delta-kusto/releases/tag/0.10.4.111)
